integration-tests/testsetups/keeper_benchmark.go

```diff
@@ -0,0 +1,357 @@
+	"github.com/smartcontractkit/chainlink-env/environment"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink-testing-framework/contracts/ethereum"
+	reportModel "github.com/smartcontractkit/chainlink-testing-framework/testreporters"
+	"github.com/smartcontractkit/chainlink/integration-tests/actions"
+	"github.com/smartcontractkit/chainlink/integration-tests/client"
+	"github.com/smartcontractkit/chainlink/integration-tests/contracts"
+	"github.com/smartcontractkit/chainlink/integration-tests/testreporters"
+)
+
+// KeeperBenchmarkTest builds a test to check that chainlink nodes are able to upkeep a specified amount of Upkeep
+// contracts within a certain block time
+type KeeperBenchmarkTest struct {
+	Inputs       KeeperBenchmarkTestInputs
+	TestReporter testreporters.KeeperBenchmarkTestReporter
+
+	keeperRegistries        []contracts.KeeperRegistry
+	keeperConsumerContracts [][]contracts.KeeperConsumerBenchmark
+	upkeepIDs               [][]*big.Int
+
+	env            *environment.Environment
+	chainlinkNodes []*client.Chainlink
+	chainClient    blockchain.EVMClient
+}
+
+// KeeperBenchmarkTestInputs are all the required inputs for a Keeper Benchmark Test
+type KeeperBenchmarkTestInputs struct {
+	BlockchainClient       blockchain.EVMClient              // Client for the test to connect to the blockchain with
+	NumberOfContracts      int                               // Number of upkeep contracts
+	KeeperRegistrySettings *contracts.KeeperRegistrySettings // Settings of each keeper contract
+	Timeout                time.Duration                     // Timeout for the test
+	BlockRange             int64                             // How many blocks to run the test for
+	BlockInterval          int64                             // Interval of blocks that upkeeps are expected to be performed
+	CheckGasToBurn         int64                             // How much gas should be burned on checkUpkeep() calls
+	PerformGasToBurn       int64                             // How much gas should be burned on performUpkeep() calls
+	ChainlinkNodeFunding   *big.Float                        // Amount of ETH to fund each chainlink node with
+	UpkeepGasLimit         int64                             // Maximum gas that can be consumed by the upkeeps
+	UpkeepSLA              int64                             // SLA in number of blocks for an upkeep to be performed once it becomes eligible
+	FirstEligibleBuffer    int64                             // How many blocks to add to randomised first eligible block, set to 0 to disable randomised first eligible block
+	RegistryVersions       []ethereum.KeeperRegistryVersion  // Registry version to use
+	PreDeployedConsumers   []string                          // PreDeployed consumer contracts to re-use in test
+	UpkeepResetterAddress  string
+	InitialBatchReset      bool
```

Comment:
I feel this is  overcomplicating the test config. Do we really need both options?